tensorflow/core/distributed_runtime/rpc/grpc_worker_service.cc

```diff
@@ -392,6 +392,24 @@ void GrpcWorker::GrpcRecvTensorAsync(CallOptions* opts,
       });
 }
 
+void GrpcWorker::LoggingAsync(const LoggingRequest* request,
+                              LoggingResponse* response, StatusCallback done) {
+  auto env = this->env();
+  if (env) {
+    auto session_mgr = (SessionMgr*)env->session_mgr;
+    if (session_mgr) {
+      session_mgr->SetLogging(request->rpc_logging());
+      if (request->clear()) {
+        session_mgr->ClearLogs();
+      }
```

Comment:
If both `clear` and `fetch_step_id` are used the semantics are unclear, but I would have thought it made more sense to retrieve any previously collected data before clearing?

Reply:
Great observation. Fixed.